### ISSUE TITLE
`dask/tests/test_distributed.py::test_default_scheduler_on_worker` failing on `main`

### ISSUE THREAD

**Issue by jrbourbeau** [dask/dask#10144]
`dask/tests/test_distributed.py::test_default_scheduler_on_worker` has started failing on `main` and PRs. See [this build](https://github.com/dask/dask/actions/runs/4607248218/jobs/8141532586?pr=10140) for an example. Testing locally, it appears https://github.com/dask/distributed/pull/7656 is where things started failing.

cc @gjoseph92 @ntabris @fjetter for visibility

**Comment by jrbourbeau** [comment]
Looks like that test is relying on computation code collection, which was turned off by default in https://github.com/dask/distributed/pull/7656

**Comment by jrbourbeau** [comment]
Naively I thought setting `distributed.diagnostics.computations.nframes` to be non-zero (like was done to test in https://github.com/dask/distributed/pull/7656) would have fixed things

```diff
diff --git a/dask/tests/test_distributed.py b/dask/tests/test_distributed.py
index 9ae29d14e..4c8ecf2da 100644
--- a/dask/tests/test_distributed.py
+++ b/dask/tests/test_distributed.py
@@ -217,8 +217,9 @@ def test_default_scheduler_on_worker(c, computation, use_distributed, scheduler)

         return True

-    res = c.submit(foo)
-    assert res.result() is True
+    with dask.config.set({"distributed.diagnostics.computations.nframes": 2}):
+        res = c.submit(foo)
+        assert res.result() is True
     # Count how many submits/update-graph were received by the scheduler
     assert (
         c.run_on_scheduler(
```

However, these tests are still failing. What am I missing?

**Comment by gjoseph92** [comment]
> However, these tests are still failing. What am I missing?

That seems odd. I can take a look at that. Nonetheless, using it as a proxy for the number of `update-graph` calls doesn't seem great; I like the plugin approach regardless for the explicit-ness.

**Comment by jrbourbeau** [comment]
> I like the plugin approach regardless for the explicit-ness.

👍  -- see https://github.com/dask/dask/pull/10145 for that 

Re: the config option not working in this test. I wouldn't look into it further in order to get the test fixed (things are passing in https://github.com/dask/dask/pull/10145). But if my understanding if correct, and this is pointing to a possible bug, then an extra set of eyes would be appreciated

**Comment by gjoseph92** [comment]
Yeah, I'd just like to quickly look into why the config didn't work, in case that's a bug.

**Comment by gjoseph92** [comment]
Ok, I think that fix didn't work because it's using the `client` fixture to create a cluster, which launches workers in subprocesses. Since this test is about calling `compute` within tasks, the tasks are running in worker subprocesses, where `distributed.diagnostics.computations.nframes` isn't set.

**Comment by jrbourbeau** [comment]
Ah, I see. That makes sense, thanks for looking into things @gjoseph92
